src/overlays/actors/ovl_En_Fish2/z_en_fish2.c

```diff
@@ -22,95 +65,1112 @@ const ActorInit En_Fish2_InitVars = {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void func_80B2A448(EnFish2* this) {
+    func_80B28370(this, 4);
+    this->unk_2B4 = 0;
+    this->unk_2C4 = 0;
+    this->unk_2B6 = this->unk_2B4;
+    D_80B2B2E4 = 2;
+    this->actionFunc = func_80B2A498;
+}
+
+void func_80B2A498(EnFish2* this, GlobalContext* globalCtx) {
+    f32 currentFrame = this->skelAnime.animCurrentFrame;
+    Vec3f sp80;
+
+    if ((this->unk_2AC == 4) && func_801378B8(&this->skelAnime, 13.0f)) {
+        Actor* temp_v0;
+
+        Math_Vec3f_Copy(&sp80, &this->unk_318);
+        temp_v0 = Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_COL_MAN, sp80.x, sp80.y, sp80.z, 0,
+                              this->actor.world.rot.y, 0, 0);
+        if (temp_v0 != NULL) {
+            temp_v0->speedXZ = 4.0f;
+            temp_v0->velocity.y = 15.0f;
+            Audio_PlayActorSound2(&this->actor, NA_SE_SY_PIECE_OF_HEART);
+            gSaveContext.weekEventReg[81] &= 0xEF;
+            gSaveContext.weekEventReg[81] &= 0xDF;
+            gSaveContext.weekEventReg[81] &= 0xBF;
+            gSaveContext.weekEventReg[81] &= 0x7F;
+            gSaveContext.weekEventReg[82] &= 0xFE;
+            gSaveContext.weekEventReg[82] &= 0xFD;
+        }
+    }
+
+    if ((this->unk_2AC == 4) && (func_801378B8(&this->skelAnime, 13.0f) || func_801378B8(&this->skelAnime, 31.0f))) {
+        WaterBox* sp78;
+
+        if (func_800CA1AC(globalCtx, &globalCtx->colCtx, this->actor.world.pos.x, this->actor.world.pos.z,
+                          &this->unk_334, &sp78)) {
+            Vec3f sp6C;
+            s32 i;
+
+            Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 50, NA_SE_EV_BOMB_DROP_WATER);
+
+            for (i = 0; i < 10; i++) {
+                Math_Vec3f_Copy(&sp6C, &this->actor.world.pos);
+                sp6C.x += randPlusMinusPoint5Scaled(70.0f);
+                sp6C.y = this->unk_334 + 10.0f;
+                sp6C.z += randPlusMinusPoint5Scaled(70.0f);
+                EffectSsGSplash_Spawn(globalCtx, &sp6C, NULL, NULL, 0, (s32)randPlusMinusPoint5Scaled(50.0f) + 0x15E);
+            }
+        }
+    }
+
+    if ((this->unk_2CC <= currentFrame) && (this->unk_2AC == 4)) {
+        D_80B2B2E0 = 0;
+        D_80B2B2E4 = 3;
+        this->actor.world.pos.x = this->unk_324.x;
+        this->actor.world.pos.z = this->unk_324.z;
+        func_80B28370(this, 0);
+    }
+}
+
+void EnFish2_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    static f32 D_80B2B3A8[] = {
+        0.0f, 40.0f, -40.0f, 0.0f, 0.0f, 0.0f,
+    };
+    EnFish2* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+
+    if ((this->actionFunc != func_80B295A4) && (this->actor.params != 1)) {
+        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    }
+
+    if (this->unk_2B8 != 0) {
+        this->unk_2B8--;
+    }
+
+    if (this->unk_2B6 != 0) {
+        this->unk_2B6--;
+    }
+
+    if (this->unk_2B4 == 0) {
+    } else {
+        this->unk_2B4--;
+    }
```

Comment:
Does this _has_ to be written like that? Can't be rewritten?
```suggestion
    if (this->unk_2B4 != 0) {
        this->unk_2B4--;
    }
```
If it doesn't match, maybe throwing a `if(1) {}` may help

Reply:
It has an unconditional branch, so it has to be like this.